 ###  Winter 2022 Data Science Intern Challenge 

### Question 1 :   


 ####  Importing the required libraries , then the dataset :

In [3]:
import pandas as pd
import matplotlib.pyplot as plt

In [4]:
data =  pd.read_csv('2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv')

In [5]:
data.shape

(5000, 7)

In [6]:
data.head(5)

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11

In [7]:
data.dtypes

order_id           int64
shop_id            int64
user_id            int64
order_amount       int64
total_items        int64
payment_method    object
created_at        object
dtype: object

#### Exploratory Data Analysis

The formula of AOV is :  Revenue / Number of Orders , So I decided to understand the behaviour of the related features of the dataset with the formula . 
Revenue = the sum of 'order amount' 
Number of Orders = number of rows of our dataset 

In [8]:
data['order_amount'].mean()

3145.128

It looks like the value calculated is the mean of the 'order_amount' of all shops , Our goal is to calculate the average order value , which means the average amount spent each time a customer places an order  

##### So it's better to evaluate the AOV by calculating  it for each shop  : 


In [9]:
shop_aov=data.groupby(['shop_id'])['order_amount'].mean().sort_values(ascending=False)
shop_aov=pd.DataFrame({'shop_id':shop_aov.index,'aov':shop_aov.values})
shop_aov['aov']=shop_aov['aov'].apply(round)
shop_aov

shop_id     aov
0        42  235101
1        78   49213
2        50     404
3        90     403
4        38     391
..      ...     ...
95       53     214
96      100     214
97       32     190
98        2     174
99       92     163

[100 rows x 2 columns]

It seems that the AOV of the two shops 78 , 42 is very high compared to the other shops . So let's display the data related to these two shops to figure out what is going on 

In [10]:
data[data['shop_id'] == 42].head(5)

order_id  shop_id  user_id  order_amount  total_items payment_method  \
15         16       42      607        704000         2000    credit_card   
40         41       42      793           352            1    credit_card   
60         61       42      607        704000         2000    credit_card   
308       309       42      770           352            1    credit_card   
409       410       42      904           704            2    credit_card   

              created_at  
15    2017-03-07 4:00:00  
40   2017-03-24 14:15:41  
60    2017-03-04 4:00:00  
308  2017-03-11 18:14:39  
409  2017-03-04 14:32:58

In [11]:
data[data['shop_id'] == 78].head(5)

order_id  shop_id  user_id  order_amount  total_items payment_method  \
160       161       78      990         25725            1    credit_card   
490       491       78      936         51450            2          debit   
493       494       78      983         51450            2           cash   
511       512       78      967         51450            2           cash   
617       618       78      760         51450            2           cash   

              created_at  
160   2017-03-12 5:56:57  
490  2017-03-26 17:08:19  
493  2017-03-16 21:39:35  
511   2017-03-09 7:23:14  
617  2017-03-18 11:18:42

It is mentionned that the price of sneakers is affordable , but what is shown above is  the opposite , So let's assume that the price is not converted from cent to dollar 
The new AOV values of the two shops are : 

-Shop 78 : 235,101 Dollar

-shop 42 : 49,213 

In [12]:
shop_aov['aov'][0]=shop_aov['aov'][0] / 100 
shop_aov['aov'][1]=shop_aov['aov'][1] /100 

In [13]:
shop_aov

shop_id   aov
0        42  2351
1        78   492
2        50   404
3        90   403
4        38   391
..      ...   ...
95       53   214
96      100   214
97       32   190
98        2   174
99       92   163

[100 rows x 2 columns]

Let's save our output into a csv file : 

In [15]:
shop_aov.to_csv('shop_aov.csv')